src/rmkit/fb/fb.cpy

```diff
@@ -248,6 +251,12 @@ namespace framebuffer:
     //
     // note that dithering does not work with GRAY, RUBBER or ERASER
     inline void draw_rect(int o_x, o_y, w, h, color, fill=true, float dither=1.0):
+      update_dirty(dirty_area, o_x, o_y)
+      update_dirty(dirty_area, o_x+w, o_y+h)
+
+      _draw_rect_fast(o_x, o_y, w, h, color, fill, dither)
+
+    inline void _draw_rect_fast(int o_x, o_y, w, h, color, fill=true, float dither=1.0):
```

Comment:
I do see a huge reduction in the number of times update_dirty is called, which seems like a net benefit in any case, so this looks like a helpful part for sure.